src/contexts/socket.context.tsx

```diff
@@ -79,6 +81,13 @@ function SocketProvider({ children }: SocketProviderProps) {
         );
       });
 
+      state.socket.on("build_deleted", function (build: Build) {
+        clearBuild(buildDispatch, build.id);
```

Comment:
```suggestion
        buildDispatch({ type: "delete", payload: build.id });
```

Reply:
let's instead directly dispatch actions, if it doesn't contain any api calls or additional logic